Sources/CobSpec/Configuration/EnumCollection.swift

```diff
@@ -0,0 +1,28 @@
+public protocol EnumCollection: Hashable {
+
+    static func cases() -> AnySequence<Self>
+    static var allValues: [Self] {get}
+
+}
+
+public extension EnumCollection {
+
+    public static func cases() -> AnySequence<Self> {
+        return AnySequence { () -> AnyIterator<Self> in
+            var raw = 0
+            return AnyIterator {
+                let current: Self = withUnsafePointer(to: &raw) {$0.withMemoryRebound(to: self, capacity: 1){$0.pointee}}
```

Comment:
Can you explain what this is doing and how it's doing it? Is this protocol/extension taken from somewhere?

Reply:
This was used in lieu of `CaseIterable`, which is a feature in Swift 4.2. I found this protocol to do the work. Do I have full comprehension of this protocol, I do not, but I can do some additional research.

Reply:
👍 Sound good. I find a lot of the code I find online can be a lot more understandable just changing some names. So that's always something to think about, especially since your function can be context dependent (unless several places need to use this EnumCollection), why not call it something like `RouteCollection`?